# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/base/glibc/glibc23/sparc32-sysdeps-2004-05-15.patch
# ROCK Linux is Copyright (C) 1998 - 2005 Clifford Wolf
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

--- libc/sysdeps/unix/sysv/linux/sparc/sparc32/sysdep.h	2003/08/31 17:23:00	1.13
+++ libc/sysdeps/unix/sysv/linux/sparc/sparc32/sysdep.h	2004/05/17 18:42:44	1.14
@@ -1,4 +1,4 @@
-/* Copyright (C) 1997, 2002, 2003 Free Software Foundation, Inc.
+/* Copyright (C) 1997, 2002, 2003, 2004 Free Software Foundation, Inc.
    This file is part of the GNU C Library.
    Contributed by Miguel de Icaza <miguel@gnu.ai.mit.edu>, January 1997.
 
@@ -70,18 +70,18 @@
 
 #if RTLD_PRIVATE_ERRNO
 # define SYSCALL_ERROR_HANDLER						\
-	.section .gnu.linkonce.t.__sparc.get_pic.l7,"ax",@progbits;	\
-	.globl __sparc.get_pic.l7;					\
-	.hidden __sparc.get_pic.l7;					\
-	.type __sparc.get_pic.l7,@function;				\
-__sparc.get_pic.l7:							\
+	.section .gnu.linkonce.t.__sparc_get_pic_l7,"ax",@progbits;	\
+	.globl __sparc_get_pic_l7;					\
+	.hidden __sparc_get_pic_l7;					\
+	.type __sparc_get_pic_l7,@function;				\
+__sparc_get_pic_l7:							\
 	retl;								\
 	 add	%o7, %l7, %l7;						\
 	.previous;							\
 SYSCALL_ERROR_HANDLER_ENTRY(__syscall_error_handler)			\
 	save	%sp,-96,%sp;						\
 	sethi	%hi(_GLOBAL_OFFSET_TABLE_-4), %l7;			\
-	call	__sparc.get_pic.l7;					\
+	call	__sparc_get_pic_l7;					\
 	 add	%l7, %lo(_GLOBAL_OFFSET_TABLE_+4), %l7;			\
 	ld	[%l7 + errno], %l0;					\
 	st	%i0, [%l0];						\
@@ -96,11 +96,11 @@
 # endif
 # ifdef SHARED
 #  define SYSCALL_ERROR_HANDLER						\
-	.section .gnu.linkonce.t.__sparc.get_pic.l7,"ax",@progbits;	\
-	.globl __sparc.get_pic.l7;					\
-	.hidden __sparc.get_pic.l7;					\
-	.type __sparc.get_pic.l7,@function;				\
-__sparc.get_pic.l7:							\
+	.section .gnu.linkonce.t.__sparc_get_pic_l7,"ax",@progbits;	\
+	.globl __sparc_get_pic_l7;					\
+	.hidden __sparc_get_pic_l7;					\
+	.type __sparc_get_pic_l7,@function;				\
+__sparc_get_pic_l7:							\
 	retl;								\
 	 add	%o7, %l7, %l7;						\
 	.previous;							\
@@ -108,7 +108,7 @@
 	save	%sp,-96,%sp;						\
 	sethi	%tie_hi22(SYSCALL_ERROR_ERRNO), %l1;			\
 	sethi	%hi(_GLOBAL_OFFSET_TABLE_-4), %l7;			\
-	call	__sparc.get_pic.l7;					\
+	call	__sparc_get_pic_l7;					\
 	 add	%l7, %lo(_GLOBAL_OFFSET_TABLE_+4), %l7;			\
 	add	%l1, %tie_lo10(SYSCALL_ERROR_ERRNO), %l1;		\
 	ld	[%l7 + %l1], %l1, %tie_ld(SYSCALL_ERROR_ERRNO);		\
